### DISCUSSION TITLE
Fragment update issue with a single quote as the content of <a> like: John's

### DISCUSSION THREAD

**Discussion by szabeszg** [unpoly/unpoly#505] [Q&A]
Hello,

The problematic fragment updated contains something like: `<a href="#" > John's apple.</a>`

If such a fragment is updated using ` up.submit($form, {target: 'main.site-main', layer: targetLayer, size: "large"});` then the following SyntaxError is shown:

```
Uncaught SyntaxError: missing ) after argument list
    at fixScriptish (unpoly.js?v=1685972703:1:104402)
    at NodeList.forEach (<anonymous>)
    at t.finalizeElement (unpoly.js?v=1685972703:1:93570)
    at afterInsert (unpoly.js?v=1685972703:1:19921)
    at Object.morph (unpoly.js?v=1685972703:1:144524)
    at t.executeStep (unpoly.js?v=1685972703:1:20294)
    at unpoly.js?v=1685972703:1:19140
    at Array.map (<anonymous>)
    at t.execute (unpoly.js?v=1685972703:1:19127)
    at t.executePlan (unpoly.js?v=1685972703:1:7274)
```

Interestingly, "normal" fragment updates which happen by normal page navigations do not result in this issue. I replaced the single quote with an apostrophe character but our site admins might use single quotes any time.

Any ideas what I could do to prevent this issue from happening?

Note that the issue does not break functionality, it is just that I dislike seeing console errors which I can't fix.

Thanks in advance!

**Comment by adam12** [comment]
Would you mind confirming your Unpoly version?

**Comment by szabeszg** [comment]
Sure, sorry for forgetting that. I kept thinking I need to specify the version number I'm using but managed to forget it in the end.

I'm using the latest release 3.2.0 / 2023-05-25

Also note that by Unicode terms: "Right Single Quotation Mark U+2019" works, while "Apostrophe" U+0027 does not.

**Comment by triskweline** [comment]
The [`fixScriptish()`](https://github.com/unpoly/unpoly/blob/8e06caceaad75925f5d967eb4fc6da4f0d91ba8f/src/unpoly/element.js#L709-L732) function in the stack trace recreates `<script>` and `<noscript>` elements from their current HTML.

I don't immediately see how the linked function could possibly throw a `SyntaxError: missing ) after argument list`, which sounds like broken JavaScript somewhere. I wonder if the single quote is a red herring, and the fragment contains a `<script>` tag or JS event handler with broken syntax somewhere.
